src/charm.py

```diff
@@ -90,7 +96,13 @@ def catch_charm_errors(func: Callable[[CharmT, EventT], None]) -> Callable[[Char
     """
 
     @functools.wraps(func)
-    def func_with_catch_errors(self, event: EventT) -> None:
+    # flake8 thinks the event argument description is missing in the docstring.
+    def func_with_catch_errors(self: CharmT, event: EventT) -> None:
```

Comment:
Is a type definition on self required?

Reply:
Yes, mypy will error with the following error if the type is removed.
```
src/charm.py:100: error: Function is missing a type annotation for one or more arguments  [no-untyped-def]
```

Reply:
Hmm, perhaps because of the decorator